lib/booker.rb

```diff
@@ -0,0 +1,95 @@
+require "time"
+require "date"
+
+require_relative "room"
+require_relative "reserve"
+require_relative "date_range"
+require_relative "blocks"
+
+class RoomBooker < Date
+  attr_reader :reservations, :rooms, :get_available_rooms
+
+  def initialize(rooms:)
+    @rooms = rooms
+    @reservations = []
+    @blocked_reservations = []
+  end
+
+  def book_reservation(check_in:, check_out:)
+    date_request = DateRange.new(check_in: check_in, check_out: check_out)
```

Comment:
It looks like you just kind of create and then throw away `DateRange` objects. Are they really needed ?